langchain4j-core/src/main/java/dev/langchain4j/internal/RetryUtils.java

```diff
@@ -190,9 +191,11 @@ public <T> T withRetry(Callable<T> action, int maxAttempts) {
             while (true) {
                 try {
                     return action.call();
+                } catch (UnrecoverableException e) {
```

Comment:
When the `ExceptionMapper` generates an `UnrecoverableException`, then the retry policy simply gives up and rethrow it as it is.